src/validation.cpp

```diff
@@ -2336,8 +2345,18 @@ bool ActivateBestChain(CValidationState &state, const CChainParams& chainparams,
             fInitialDownload = IsInitialBlockDownload();
 
             for (const PerBlockConnectTrace& trace : connectTrace.GetBlocksConnected()) {
-                assert(trace.pblock && trace.pindex);
-                GetMainSignals().BlockConnected(trace.pblock, trace.pindex, *trace.conflictedTxs);
+                assert(trace.pindex);
+                std::shared_ptr<const CBlock> pthisBlock;
+                if (trace.pblock == NULL) {
+                    std::shared_ptr<CBlock> pblockNew = std::make_shared<CBlock>();
```

Comment:
probably no need to make pblockNew? Can just assign directly with make_shared to pthisBlock.